packages/purpose-process/src/services/purposeService.ts

```diff
@@ -596,6 +607,153 @@ export function purposeServiceBuilder(
         totalCount: purposesList.totalCount,
       };
     },
+    async createPurposeVersion({
+      purposeId,
+      seed,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      purposeId: PurposeId;
+      seed: ApiPurposeVersionSeed;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<PurposeVersion> {
+      logger.info(`Creating Version for Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
+      assertDailyCallsIsDifferentThanBefore(purpose.data, seed.dailyCalls);
+
+      const conflictVersion = purpose.data.versions.find(
+        (v) =>
+          v.state === purposeVersionState.draft ||
+          v.state === purposeVersionState.waitingForApproval
+      );
+
+      if (conflictVersion !== undefined) {
+        throw purposeVersionStateConflict(
+          purposeId,
+          conflictVersion.id,
+          conflictVersion.state
+        );
+      }
+
+      const eservice = await retrieveEService(
+        purpose.data.eserviceId,
+        readModelService
+      );
+
+      function buildUpdatedPurpose({
+        newPurposeVersion,
+        archiveOldVersions,
+      }: {
+        newPurposeVersion: PurposeVersion;
+        archiveOldVersions: boolean;
+      }): Purpose {
+        const oldVersions = purpose.data.versions
+          // Remove all versions in waiting for approval state
+          .filter((v) => v.state !== purposeVersionState.waitingForApproval)
```

Comment:
Is this actually needed? Few lines above you are already throwing an error in case of any `waitingForApproval` versions.

Reply:
You are right, it's not needed there!